core/src/test/java/org/infinispan/container/SimpleDataContainerTest.java

```diff
@@ -283,4 +291,41 @@ public void testGetDuringKeySetLoop() {
 
       assertEquals(10, i);
    }
+
+   public void testEntrySetStreamWithExpiredEntries() {
```

Comment:
Since the set is CONCURRENT, I wonder if it would make sense to create a test that would add/remove entries in the middle of the toArray() call.